website/blog/2018-04-04-using-typescript-with-react-native.md

```diff
@@ -0,0 +1,310 @@
+            <Button
+              title="-"
+              onPress={this.state.onDecrement}
+              accessibilityLabel="decrement"
+              color="red"
+            />
+          </View>
+
+          <View style={styles.button}>
+            <Button
+              title="+"
+              onPress={this.state.onIncrement}
+              accessibilityLabel="increment"
+              color="blue"
+            />
+          </View>
+        </View>
+      </View>
+    );
+  }
+}
+
+// styles
+const styles = StyleSheet.create({
+  root: {
+    alignItems: 'center',
+    alignSelf: 'center',
+  },
+  buttons: {
+    flexDirection: 'row',
+    minHeight: 70,
+    alignItems: 'stretch',
+    alignSelf: 'center',
+    borderWidth: 5,
+  },
+  button: {
+    flex: 1,
+    paddingVertical: 0,
+  },
+  greeting: {
+    color: '#999',
+    fontWeight: 'bold',
+  },
+});
+```
+
+Whoa! That's a lot, but let's break it down:
+
+* Instead of rendering HTML elements like `div`, `span`, `h1`, etc., we're rendering components like `View` and `Button`. These are native components that work across different platforms.
+* Styling is specified using the `StyleSheet.create` function that React Native gives us. React's stylesheets allow us to control our layout using Flexbox, and style using other constructs similar to those in CSS.
+
+## Adding a Component Test
+
+Now that we've got a component, let's try testing it.
+
+We already have Jest installed as a test runner. We're going to write snapshot tests for our components, let's add the required add-on for snapshot tests:
+
+```sh
+yarn add --dev react-addons-test-utils
+```
+
+Now let's create a `__tests__` folder in the `components` directory and add a test for `Hello.tsx`:
+
+```ts
+// components/__tests__/Hello.tsx
+import * as React from 'react';
+import * as renderer from 'react-test-renderer';
+
+import {Hello} from '../Hello';
+
+it('renders correctly with defaults', () => {
+  const button = renderer
+    .create(<Hello name="World" enthusiasmLevel={1} />)
+    .toJSON();
+  expect(button).toMatchSnapshot();
+});
+```
+
+The first time the test is ran, it will create a snapshot of the DOM and store it in the `components/__tests__/__snapshots__/Hello.tsx.snap` file. When you modify your component, you'll need to update the snapshots and review the update for inadvertent changes. You can read more about testing React Native components [here](https://facebook.github.io/jest/docs/en/tutorial-react-native.html).
+
+## Next Steps
+
+Check out [our React tutorial](https://github.com/DanielRosenwasser/React-TypeScript-Tutorial) where we also cover topics like state management with [Redux](http://redux.js.org). These same subjects can be applied when writing React Native apps.
+
+Additionally, you may want to look at the [ReactXP](https://microsoft.github.io/reactxp/) if you're looking for a component library written entirely in TypeScript that supports both React on the web as well as React Native.
+
```

Comment:
Think both of these could maybe be ported to talk about blog posts on the RN blog, or to change the "our" to "Microsofts" above